File: Sample Code/DJISdkDemo/Demo/Camera/CameraActionsTableViewController.m
//
//  CameraActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//

#import <DJISDK/DJISDK.h>
#import "DemoUtility.h"
#import "CameraActionsTableViewController.h"
#import "CameraPushInfoViewController.h"
#import "CameraISOViewController.h"
#import "CameraShootSinglePhotoViewController.h"
#import "CameraRecordVideoViewController.h"
#import "CameraPlaybackPushInfoViewController.h"
#import "CameraPlaybackCommandViewController.h"
#import "CameraPlaybackDownloadViewController.h"
#import "CameraFetchMediaViewController.h"
#import "CameraFPVViewController.h"

@interface CameraActionsTableViewController ()

@end

@implementation CameraActionsTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.sectionNames = [NSMutableArray arrayWithArray:@[@"General", @"FPV", @"Shoot Photo", @"Record Video", @"Playback", @"Media Download"]];
    
    // General
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Push Info" andClass:[CameraPushInfoViewController class]],
                            [DemoSettingItem itemWithName:@"Set/Get ISO" andClass:[CameraISOViewController class]]]];
    // FPV
    [self.items addObject:@[[DemoSettingItem itemWithName:@"First Person View (FPV)" andClass:[CameraFPVViewController class]]]];
    
    // Shoot Photo
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Shoot Single Photo" andClass:[CameraShootSinglePhotoViewController class]]]];
    
    // Record Video
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Record video" andClass:[CameraRecordVideoViewController class]]]];
    
    // Playback
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Playback Push Info" andClass:[CameraPlaybackPushInfoViewController class]],
                            [DemoSettingItem itemWithName:@"Playback commands" andClass:[CameraPlaybackCommandViewController class]],
                            [DemoSettingItem itemWithName:@"Playback Download" andClass:[CameraPlaybackDownloadViewController class]]]];

    // Media Download
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Fetch media" andClass:[CameraFetchMediaViewController class]]]];
    
}

-(DJIBaseComponent *)getComponent {
    return [DemoComponentHelper fetchCamera];
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/Camera/CameraActionsTableViewController.swift
//
//  CameraActionsTableViewController.h
//  DJISdkDemo
//
//  Created by DJI on 12/18/15.
//  Copyright © 2015 DJI. All rights reserved.
//

import DJISDK
class CameraActionsTableViewController: DemoTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.connectedComponent = ConnectedProductManager.sharedInstance.fetchCamera()
        self.showComponentVersionSn = true
        
        self.sectionNames = ["General", "FPV", "Shoot Photo", "Record Video", "Playback", "Media Download"]
        // General
        self.items.append([DemoSettingItem(name: "Push Info", andClass: CameraPushInfoViewController.self), DemoSettingItem(name: "Set/Get ISO", andClass: CameraISOViewController.self )])
        
        // FPV
        self.items.append([DemoSettingItem(name: "First Person View (FPV)", andClass: CameraFPVViewController.self)])
        
        // Shoot Photo
        self.items.append([DemoSettingItem(name: "Shoot Single Photo", andClass: CameraShootSinglePhotoViewController.self)])
        
        // Record Video
        self.items.append([DemoSettingItem(name: "Record video", andClass: CameraRecordVideoViewController.self)])
        
        // Playback
        self.items.append([DemoSettingItem(name: "Playback Push Info", andClass: CameraPlaybackPushInfoViewController.self), DemoSettingItem(name: "Playback commands", andClass: CameraPlaybackCommandViewController.self), DemoSettingItem(name: "Playback Download", andClass: CameraPlaybackDownloadViewController.self)])
       
        // Media Download
        self.items.append([DemoSettingItem(name: "Fetch media", andClass: CameraFetchMediaViewController.self)])
    }
}
